File: Server/src/main/java/com/example/server/entity/Configuration.java
package com.example.server.entity;

import jakarta.persistence.*;
import lombok.AllArgsConstructor;
import lombok.Data;
import lombok.NoArgsConstructor;
import lombok.NonNull;

/**
 * Entity representing the configuration settings for the ticketing system.
 * This class stores the configuration details that determine the behavior of the system,
 * such as ticket capacity, release and retrieval intervals, and ticket management rules.
 */
@AllArgsConstructor
@NoArgsConstructor
@Data
@Entity
public class Configuration {
    @Id
    @GeneratedValue(strategy = GenerationType.IDENTITY)
    private long id;

    @NonNull
    @Column(nullable = false)
    private String name;

    @Column(nullable = false)
    private int maxTicketCapacity;

    @Column(nullable = false)
    private int totalTickets;

    @Column(nullable = false)
    private int ticketsPerRelease;

    @Column(nullable = false)
    private int releaseIntervalMilliseconds;

    @Column(nullable = false)
    private int ticketsPerRetrieval;

    @Column(nullable = false)
    private int retrievalIntervalMilliseconds;


}

File: client/src/app/model/class/Configuration.ts
/**
 * Represents a configuration for ticket management in the system.
 * Stores information related to ticket capacity, release, and retrieval settings.
 */
export class Configuration{
    /** Unique identifier for the configuration. */
    id: number;

    /** Name of the configuration. */
    name: string;

    /** Maximum capacity for tickets in the system. */
    maxTicketCapacity: number;

    /** Total number of tickets in the configuration. */
    totalTickets: number;

    /** Number of tickets to be released at each interval. */
    ticketsPerRelease: number;

    /** Time interval (in milliseconds) between each release of tickets. */
    releaseIntervalMilliseconds: number;

    /** Number of tickets to be retrieved at each interval. */
    ticketsPerRetrieval: number;

    /** Time interval (in milliseconds) between each retrieval of tickets. */
    retrievalIntervalMilliseconds: number;

    /**
     * Initializes a new Configuration instance with default values.
     */
    constructor() {
        this.id = 0;
        this.name = "";
        this.maxTicketCapacity = 0;
        this.totalTickets = 0;
        this.ticketsPerRelease = 0;
        this.releaseIntervalMilliseconds = 0;
        this.ticketsPerRetrieval = 0;
        this.retrievalIntervalMilliseconds = 0;
    }
}